vcd/resource_vcd_vapp_vm.go

```diff
@@ -668,8 +668,28 @@ func addVdcNetwork(networkNameToAdd string, vdc *govcd.Vdc, vapp govcd.VApp, vcd
 	return vdcNetwork, nil
 }
 
-// isItVappNetwork checks if it is a genuine vApp network (not only attached to vApp)
+// isItVappNetwork checks if it is a vApp network (not vApp Org Network)
 func isItVappNetwork(vAppNetworkName string, vapp govcd.VApp) (bool, error) {
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return false, fmt.Errorf("error getting vApp networks: %s", err)
+	}
+
+	for _, networkConfig := range vAppNetworkConfig.NetworkConfig {
+		if networkConfig.NetworkName == vAppNetworkName &&
+			networkConfig.Configuration.FenceMode == types.FenceModeIsolated ||
+			(networkConfig.Configuration.FenceMode == types.FenceModeNAT &&
+				networkConfig.Configuration.IPScopes.IPScope[0].IsInherited == false) {
```

Comment:
We should check for `networkConfig.Configuration.IPScopes` to be not nil , and for `networkConfig.Configuration.IPScopes.IPScope` to have elements

Reply:
Yes, and also this check needs an explanation comment, explaining what is the meaning of it.

Reply:
it always return, but added